### ISSUE TITLE
Not showing SC2086 when inside ONBUILD

### ISSUE THREAD

**Issue by matrixik** [hadolint/hadolint#230]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

Show errors for both RUNs.

### Actual behavior

```
Dockerfile:3 SC2086 Double quote to prevent globbing and word splitting.
```

### Steps to reproduce the behavior

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter v1.9.0-0-g4c4881a
```

**Dockerfile**

```Dockerfile
FROM scratch

RUN \
    echo $EXTRA_DEPS

ONBUILD RUN \
    echo $EXTRA_DEPS
```

**Additional environment details (OS, stack version, etc.)**
```
System:    Host: zybortd Kernel: 4.10.0-42-generic x86_64 (64 bit gcc: 6.3.0)
           Desktop: Gnome  (Gtk 3.22.11-0ubuntu3) Distro: Ubuntu 17.04
```

**Comment by lorenzo** [comment]
Hmm, this is really odd, I was sure rules for `ONBUILD` where being applied. Thanks for reporting!

**Comment by lorenzo** [comment]
@matrixik no, that is no currently possible, as for the parser there is no notion of multiple lines in any instruction the `\` at the end of the line is pre-processed to mean "remove the newline char"
